modules/ingester/instance_search.go

```diff
@@ -55,9 +55,17 @@ func (i *instance) Search(ctx context.Context, req *tempopb.SearchRequest) (*tem
 
 	sr.AllWorkersStarted()
 
+	// read and combine search results
 	resultsMap := map[string]*tempopb.TraceSearchMetadata{}
 
+	// collect results from all the goroutines via sr.Results channel.
+	// range loop will exit when sr.Results channel is closed.
 	for result := range sr.Results() {
+		// exit early and Propagate error upstream
+		if sr.Error() != nil {
```

Comment:
if a query returns no results and hits an error will this line be executed? Do we need a check after the for loop to see if `sr.Error()` equals anything?

Reply:
done